remote-api/src/main/java/org/alfresco/rest/api/model/rules/SimpleCondition.java

```diff
@@ -0,0 +1,238 @@
+            builder.parameter(actionCondition.getParameterValues().get(ComparePropertyValueEvaluator.PARAM_VALUE).toString());
+            break;
+        case CompareMimeTypeEvaluator.NAME:
+            builder.field(actionCondition.getParameterValues().get(ComparePropertyValueEvaluator.PARAM_PROPERTY).toString());
+            builder.comparator(COMPARATOR_EQUALS);
+            builder.parameter(actionCondition.getParameterValues().get(ComparePropertyValueEvaluator.PARAM_VALUE).toString());
+            break;
+        case HasAspectEvaluator.NAME:
+            builder.field(HasAspectEvaluator.PARAM_ASPECT);
+            builder.comparator(COMPARATOR_EQUALS);
+            builder.parameter(actionCondition.getParameterValues().get(HasAspectEvaluator.PARAM_ASPECT).toString());
+            break;
+        case HasChildEvaluator.NAME:
+            if (actionCondition.getParameterValues().get(HasChildEvaluator.PARAM_ASSOC_TYPE) != null)
+            {
+                builder.field(actionCondition.getParameterValues().get(HasChildEvaluator.PARAM_ASSOC_TYPE).toString());
+            } else {
+                builder.field(actionCondition.getParameterValues().get(HasChildEvaluator.PARAM_ASSOC_NAME).toString());
+            }
+            builder.comparator(COMPARATOR_EQUALS);
+            builder.parameter(actionCondition.getParameterValues().get(ComparePropertyValueEvaluator.PARAM_VALUE).toString());
+            break;
+        case HasTagEvaluator.NAME:
+            builder.field(HasTagEvaluator.PARAM_TAG);
+            builder.comparator(COMPARATOR_EQUALS);
+            builder.parameter(actionCondition.getParameterValues().get(HasTagEvaluator.PARAM_TAG).toString());
+            break;
+        case HasVersionHistoryEvaluator.NAME:
+            builder.field(actionCondition.getParameterValues().get(ComparePropertyValueEvaluator.PARAM_PROPERTY).toString().toLowerCase());
+            builder.comparator(actionCondition.getParameterValues().get(ComparePropertyValueEvaluator.PARAM_OPERATION).toString());
+            builder.parameter(actionCondition.getParameterValues().get(ComparePropertyValueEvaluator.PARAM_VALUE).toString());
+            break;
+        case InCategoryEvaluator.NAME:
+            builder.field(actionCondition.getParameterValues().get(InCategoryEvaluator.PARAM_CATEGORY_ASPECT).toString());
+            builder.comparator(COMPARATOR_EQUALS);
+            builder.parameter(actionCondition.getParameterValues().get(InCategoryEvaluator.PARAM_CATEGORY_VALUE).toString());
+            break;
+        case IsSubTypeEvaluator.NAME:
+            builder.field(IsSubTypeEvaluator.PARAM_TYPE);
+            builder.comparator(COMPARATOR_EQUALS);
+            builder.parameter(actionCondition.getParameterValues().get(IsSubTypeEvaluator.PARAM_TYPE).toString());
+            break;
+        case NoConditionEvaluator.NAME:
+        default:
+            return null;
+        }
+
+        return builder.create();
```

Comment:
We're not really getting neater code from the builder here.  Maybe if we had lots of methods like:
```
...
case IsSubTypeEvaluator.NAME:
    return makeIsSubTypeCondition(actionCondition);
...
private Action makeIsSubTypeCondition(actionCondition)
{
    return SimpleCondition.builder()
            .field(IsSubTypeEvaluator.PARAM_TYPE)
            .comparator(COMPARATOR_EQUALS)
            .parameter(actionCondition.getParameterValues().get(IsSubTypeEvaluator.PARAM_TYPE).toString())
            .create();
}
```

Reply:
Fixed according to your suggestion. Thanks